Makefile

```diff
@@ -200,21 +200,21 @@ integration: GODOG_OPTS = --tags=basic
 integration: $(MINISHIFT_BINARY)
 	mkdir -p $(INTEGRATION_TEST_DIR)
 	go test -timeout $(TIMEOUT) $(REPOPATH)/test/integration --tags=integration -v -args --test-dir $(INTEGRATION_TEST_DIR) --binary $(MINISHIFT_BINARY) \
-	--run-before-feature="$(RUN_BEFORE_FEATURE)" $(GODOG_OPTS)
+	--run-before-feature="$(RUN_BEFORE_FEATURE)" --test-with-shell=$(TEST_WITH_SHELL) $(GODOG_OPTS)
```

Comment:
@agajdosi `$(TEST_WITH_SHELL)` where is this coming from? I do not see how we are deciding the shell for the Makefile.

Reply:
what's the default?

Reply:
@hferentschik @LalatenduMohanty default is empty string, if empty string is being detected in shell.go, then the test will use bash if test runs on linux or mac, or powershell if the OS is windows, pretty much it imitates what Minishift uses.

When one uses TEST_WITH_SHELL, then the test can be forced to run with: cmd, powershell, bash, tcsh, zsh (fish has problems and the program will inform you about it and will run the bash).

Reply:
what's about the docs around this?

Reply:
@hferentschik As this is quite specific I have documented it directly in feature files (https://github.com/minishift/minishift/issues/1332), but I can add it into docs also.

Reply:
Well, given that it is exposed in the Makefile I would expect to get some form of information. Maybe a comment in the Makefile. The rub is that if I read the Makefile I see '$(TEST_WITH_SHELL)'. I then notice that there is no value for that at all, which slowly get my alarm bells ringing. I guess @LalatenduMohanty was in a similar boat. 

For what it's worth, the same issue is with 'RUN_BEFORE_FEATURE'. It feels wrong to always see '--run-before-feature' and '--test-with-shell', effectively knowing that these values are empty. The expectation from using command line switches is that one does not specify them, if there is no value one wants to pass. People react to that when reading the Makefile.

Reply:
@hferentschik I also feel that this solution we use already for some time is a bit strange. One simple solution would be to make these values empty with quotes, so it would be at least more clear that it is passing nothing:
```
go test -timeout 3600s [...] --run-before-feature="" --test-with-shell="" [] -tags=cmd-docker-env,cmd-oc-env
```

another possibility would be to have just one parameter, for example `TEST_OPTS` to which one would need to pass the all the flags she wants to use, probably also mixed with GODOG_OPTS, so we would just se the TEST_OPTS:

```
make integration TEST_OPTS="--test-with-shell=bash --tags=cmd-docker-env --run-before-feature='minishift config set memory 4GB;minishift config set cpus 4'"
```

WDYT?

Reply:
However for the second solution I would say it is better to create a follow up issue.